types/validator_set_test.go

```diff
@@ -62,8 +62,12 @@ func TestValidatorSetBasic(t *testing.T) {
 
 	// update
 	assert.False(t, vset.Update(randValidator_()))
-	val.VotingPower = 100
+	_, val = vset.GetByAddress(val.Address)
+	val.VotingPower += 100
+	proposerPriority := val.ProposerPriority
 	assert.True(t, vset.Update(val))
+	_, val = vset.GetByAddress(val.Address)
+	assert.Equal(t, proposerPriority, val.ProposerPriority)
```

Comment:
Interestingly, this assertion doesn't fail even if we didn't add the line that aims to preserve the validator's ProposerPriority: https://github.com/tendermint/tendermint/pull/2941/files#diff-bd488e139d955daba658159f507010f7R290
Probably, due to the update not coming from `types.PB2TM.ValidatorUpdates` where the ProposerPriority would be `0`. To make this test, test the change made, we'd need to add a line `val.ProposerPriority = 0` right before the `Update`.